core/client/src/test/java/alluxio/hadoop/AbstractFileSystemTest.java

```diff
@@ -248,6 +289,14 @@ public void getStatus() throws Exception {
     return conf;
   }
 
+  private void mockFileSystemContextAndMasterClient() {
+    mMockFileSystemContext = Mockito.mock(FileSystemContext.class);
+    Whitebox.setInternalState(FileSystemContext.class, "INSTANCE", mMockFileSystemContext);
```

Comment:
@aaudiber @calvinjia I think this current design is not unit-test friendly as we talked before. Therefore instead of keeping enlarging the footprint of this PR. we should keep it simple but have separate effort to clean up the unit tests (e.g., removing whiteboxing)